src/network/message.rs

```diff
@@ -144,14 +192,14 @@ impl<N: Network, E: Environment> Message<N, E> {
         // Deserialize the data field.
         let message = match id {
             0 => Self::BlockRequest(bincode::deserialize(&data[0..4])?, bincode::deserialize(&data[4..8])?),
-            1 => Self::BlockResponse(bincode::deserialize(data)?),
+            1 => Self::BlockResponse(MaybeSerialized::Serialized(data.to_vec())),
             2 => Self::ChallengeRequest(
                 bincode::deserialize(&data[0..4])?,
                 bincode::deserialize(&data[4..6])?,
                 bincode::deserialize(&data[6..14])?,
                 bincode::deserialize(&data[14..18])?,
```

Comment:
Would be good to comment on the various indexes used here?

Reply:
I would like to handle deserialization in a streaming fashion (which would remove the need for indexing altogether), but that's follow-up material.